src/components/side-panel/layer-panel/layer-list.js

```diff
@@ -0,0 +1,168 @@
+import React, {useCallback, useMemo, useState} from 'react';
+
+import {arrayMove} from 'utils/data-utils';
+import styled from 'styled-components';
+import classnames from 'classnames';
+import {SortableContainer, SortableElement} from 'react-sortable-hoc';
+
+import LayerPanelFactory from './layer-panel';
+
+// make sure the element is always visible while is being dragged
+// item being dragged is appended in body, here to reset its global style
+const SortableStyledItem = styled.div`
+  z-index: ${props => props.theme.dropdownWrapperZ + 1};
+
+  &.sorting {
+    pointer-events: none;
+  }
+
+  &.sorting-layers .layer-panel__header {
+    background-color: ${props => props.theme.panelBackgroundHover};
+    font-family: ${props => props.theme.fontFamily};
+    font-weight: ${props => props.theme.fontWeight};
+    font-size: ${props => props.theme.fontSize};
+    line-height: ${props => props.theme.lineHeight};
+    *,
+    *:before,
+    *:after {
+      box-sizing: border-box;
+    }
+    .layer__drag-handle {
+      opacity: 1;
+      color: ${props => props.theme.textColorHl};
+    }
+  }
+`;
+
+LayerListFactory.deps = [LayerPanelFactory];
+
+function LayerListFactory(LayerPanel) {
+  // By wrapping layer panel using a sortable element we don't have to implement the drag and drop logic into the panel itself;
+  // Developers can provide any layer panel implementation and it will still be sortable
+  const SortableItem = SortableElement(({children, isSorting}) => {
+    return (
+      <SortableStyledItem className={classnames('sortable-layer-items', {sorting: isSorting})}>
+        {children}
+      </SortableStyledItem>
+    );
+  });
+
+  const WrappedSortableContainer = SortableContainer(({children}) => {
+    return <div>{children}</div>;
+  });
+
+  const LayerList = props => {
+    const {
+      layers,
+      datasets,
+      layerOrder,
+      uiStateActions,
+      visStateActions,
+      layerClasses,
+      isSortable
+    } = props;
+    const {toggleModal: openModal} = uiStateActions;
+    const [isSorting, setIsSorting] = useState(false);
+
+    const layerTypeOptionsSelector = useCallback(
+      () =>
+        Object.keys(layerClasses).map(key => {
+          const layer = new layerClasses[key]();
+          return {
+            id: key,
+            label: layer.name,
+            icon: layer.layerIcon,
+            requireData: layer.requireData
+          };
+        }),
+      [layerClasses]
+    );
+
+    const layerTypeOptions = useMemo(() => layerTypeOptionsSelector(), [layerTypeOptionsSelector]);
+
```

Comment:
why creating `layerTypeOptionsSelector` when you only use it once?
you can simply do:
```
const layerTypeOptions = useMemo(() => Object.keys(layerClasses).map(key => {
          const layer = new layerClasses[key]();
          return {
            id: key,
            label: layer.name,
            icon: layer.layerIcon,
            requireData: layer.requireData
          };
        }), [layerClasses]);
```

Reply:
Idk, the whole code was just a copy-paste from layer-manager so I didn't think about it much, honestly.